dapp_runner/runner/__init__.py

```diff
@@ -51,27 +51,52 @@ async def _run_app(
         )
 
     assert r.commissioning_time  # sanity check for mypy
+
+    startup_timeout = timedelta(
+        seconds=config.limits.startup_timeout or DEFAULT_STARTUP_TIMEOUT
+    )
+
+    max_running_time: Optional[timedelta] = None
+    if config.limits.max_running_time:
+        max_running_time = timedelta(seconds=config.limits.max_running_time)
+
+    logger.info(
+        "Starting app: %s, startup timeout: %s, maximum running time: %s",
+        dapp.meta.name,
+        startup_timeout,
+        max_running_time,
+    )
+
+    time_started: Optional[datetime] = None
+
     try:
-        while (
-            not r.dapp_started
-            and datetime.now(timezone.utc) < r.commissioning_time + STARTING_TIMEOUT
-        ):
+        while not r.dapp_started and utcnow() < r.commissioning_time + startup_timeout:
             await asyncio.sleep(1)
 
         if not r.dapp_started:
             raise Exception(
-                f"Failed to start instances before {STARTING_TIMEOUT} elapsed."
+                f"Failed to start instances before {startup_timeout} elapsed."
             )
 
-        print(yellow("Dapp started."))
+        time_started = utcnow()
+        logger.info("Application started.")
 
-        while r.dapp_started:
+        while r.dapp_started and (
+            not max_running_time or utcnow() < time_started + max_running_time
```

Comment:
could we delegate this time check to a separate function?
Like `while r.can_run()` or `r.time_left() > 0` or other, as it would both increase readability and check below (line 93-95) could be simplified as well

Reply:
yes, I had thought about it, sure

Reply:
@VanDavv updated